File: runtime/Jlib/bllong.java
package bigloo;

public class bllong extends numeral
{
  public final long value;

  public bllong( final long  value )
  {
    this.value= value;
  }

  public static bllong make_llong( final String  s )
  {
    return new bllong( Long.parseLong( s ) );
  }

  public void write( final output_port  p )
  {
    p.write( "#l" + value );
  }
}

File: runtime/CSlib/bllong.cs
using System;

namespace bigloo
{
  public sealed class bllong: numeral
  {
    public readonly long value;

    public bllong( long  value )
    {
      this.value= value;
    }

    public static bllong make_llong( String  s )
    {
      return new bllong( Int64.Parse( s ) );
    }

    public override void write( output_port  p )
    {
      p.write( "#l" + value );
    }
  }
}
